derive/src/compile_bytecode.rs

```diff
@@ -0,0 +1,135 @@
+use super::Diagnostic;
+use bincode;
+use proc_macro2::{Span, TokenStream as TokenStream2};
+use quote::quote;
+use rustpython_compiler::{bytecode::CodeObject, compile};
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+use syn::parse::{Parse, ParseStream, Result as ParseResult};
+use syn::{self, parse2, Ident, Lit, LitByteStr, Meta, MetaList, NestedMeta, Token};
+
+struct BytecodeConst {
+    ident: Ident,
+    meta: MetaList,
+}
+
+impl BytecodeConst {
+    fn compile(&self, manifest_dir: &Path) -> Result<CodeObject, Diagnostic> {
+        let meta = &self.meta;
+
+        let mut source_path = None;
+        let mut mode = None;
+        let mut source_lit = None;
+
+        for meta in &meta.nested {
+            match meta {
+                NestedMeta::Literal(lit) => source_lit = Some(lit),
+                NestedMeta::Meta(Meta::NameValue(name_value)) => {
+                    if name_value.ident == "mode" {
+                        mode = Some(match &name_value.lit {
+                            Lit::Str(s) => match s.value().as_str() {
+                                "exec" => compile::Mode::Exec,
+                                "eval" => compile::Mode::Eval,
+                                "single" => compile::Mode::Single,
+                                _ => bail_span!(s, "mode must be exec, eval, or single"),
+                            },
+                            _ => bail_span!(name_value.lit, "mode must be a string"),
+                        })
+                    } else if name_value.ident == "source_path" {
+                        source_path = Some(match &name_value.lit {
+                            Lit::Str(s) => s.value(),
+                            _ => bail_span!(name_value.lit, "source_path must be string"),
+                        })
+                    }
+                }
+                _ => {}
+            }
+        }
+
+        let source = if meta.ident == "file" {
```

Comment:
Rather than implementing this as a rust syntax extension (which is very cool by the way;) . The code might be simpler to understand if we use a procedural macro here. The syntax extension seems overkill to me, or did I miss something?